perch/main.py

```diff
@@ -185,12 +185,18 @@ def upload_indicators_csv(ctx, indicator_file, api_key, username, password):
     headers = authenticate(ctx, api_key, username, password)
     validate_csv(ctx, indicator_file)
     community_ids = prompt_for_communities(ctx, headers)
-    req_body = []
+    req_bodies = []
     bad_rows = []
+    chunk = 0
     for index, row in enumerate(readrows(indicator_file)):
         indicator, error_message = build_indicator(row, communities=community_ids)
         if indicator:
-            req_body.append(indicator)
+            try:
+                req_bodies[chunk].append(indicator)
+            except IndexError:
+                req_bodies.append([indicator])
+            if index % INDICATOR_CHUNK_SIZE == 0:
+                chunk += 1
```

Comment:
This could also be written as:
```py
req_bodies = []
chunk = []
bad_rows = []
for index, row in enumerate(readrows(indicator_file)):
    indicator, error_message = build_indicator(row, communities=community_ids)
    if indicator:
        chunk.append(indicator)
        if len(chunk) >= INDICATOR_CHUNK_SIZE:
            req_bodies.append(chunk)
            chunk = []
    else:
        bad_rows.append({'row': index + 1, 'reason': error_message, 'data': row})

if chunk:
    req_bodies.append(chunk)
```

Reply:
This is better. Thanks!